component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1095,9 +1101,20 @@ private boolean getConnection(HttpURLConnection httpConnection, AuthenticationCo
             } else if (SMSOTPConstants.POST_METHOD.equalsIgnoreCase(httpMethod)) {
                 httpConnection.setRequestMethod(SMSOTPConstants.POST_METHOD);
                 if (StringUtils.isNotEmpty(payload)) {
-                    payload = payload.replaceAll("\\$ctx.num", encodedMobileNo).replaceAll("\\$ctx.msg", smsMessage +
-                            otpToken);
-
+                    String contentType = ((String) headerElementProperties.get(SMSOTPConstants.CONTENT_TYPE)).trim();
+                    if (!SMSOTPUtils.isPayloadEncodingForSMSOTPEnabled(context)){
+                        encodedSMSMessage = smsMessage;
+                        if (StringUtils.isNotBlank(contentType) && SMSOTPConstants.POST_METHOD.equals(httpMethod) &&
+                                (SMSOTPConstants.JSON_CONTENT_TYPE).equals(contentType)) {
+                            encodedMobileNo = receivedMobileNumber;
+                        }
+                    }
+                    else{
+                        encodedMobileNo = getEncodedValue(contentType,receivedMobileNumber);
+                        encodedSMSMessage = getEncodedValue(contentType,smsMessage);
+                    }
```

Comment:
Formatting error

Reply:
Shall we invert the if-else block? And also add a comment on how the backward compatibility is preserved here?

Reply:
Addressed with [5186befd6f64157a75c30fcd83ce4e4473e65485](https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/124/commits/5186befd6f64157a75c30fcd83ce4e4473e65485)